src/common/hooks/account/use-next-tx-nonce.ts

```diff
@@ -0,0 +1,28 @@
+import { AddressNonces } from '@stacks/blockchain-api-client/lib/generated';
+import { useGetAccountNonce } from '@common/hooks/account/use-get-account-nonce';
+import { UseQueryOptions } from 'react-query';
+import { useLastApiNonceState } from '@store/accounts/nonce.hooks';
+
+export function correctNextNonce(apiNonce: AddressNonces): number | undefined {
+  if (!apiNonce) return;
+
+  const missingNonces = apiNonce.detected_missing_nonces;
+  if (
+    missingNonces &&
+    missingNonces.length > 0 &&
+    missingNonces[0] > (apiNonce.last_executed_tx_nonce || 0)
+  ) {
+    return missingNonces.sort()[0];
+  }
+  return apiNonce.possible_next_nonce;
+}
+
+export function useNextTxNonce() {
+  const [nonce, setLastApiNonce] = useLastApiNonceState();
+  const onSuccess = (data: AddressNonces) => {
+    const nextNonce = data && correctNextNonce(data);
+    if (typeof nextNonce === 'number') setLastApiNonce(nextNonce);
+  };
+  useGetAccountNonce({ onSuccess } as UseQueryOptions);
+  return nonce;
```

Comment:
This is the only unusual looking code to me. Is there a reason not to do 

```suggestion
  const onSuccess = (data: AddressNonces) => {
    const nextNonce = data && correctNextNonce(data);
    if (typeof nextNonce === 'number') setLastApiNonce(nextNonce);
  };
  return { nonceResponse: data }  useGetAccountNonce();
  return correctNextNonce(nonceResponse);
```

Reply:
We need to use the setLastApiNonce which is used in https://github.com/blockstack/stacks-wallet-web/blob/fix/improve-nonce-handling/I934/src/store/accounts/nonce.ts#L35.
The onSuccess is needed because we need to do that only once, otherwise the react component will re-render indefinitely